paasta_tools/kubernetes/bin/paasta_secrets_sync.py

```diff
@@ -30,20 +32,35 @@
 from paasta_tools.kubernetes_tools import update_secret
 from paasta_tools.secret_tools import get_secret_provider
 from paasta_tools.utils import DEFAULT_SOA_DIR
+from paasta_tools.utils import get_service_instance_list
+from paasta_tools.utils import INSTANCE_TYPES
+from paasta_tools.utils import list_services
 from paasta_tools.utils import load_system_paasta_config
 
 
 log = logging.getLogger(__name__)
 
+INSTANCE_TYPE_TO_K8S_NAMESPACE = {}
+for instance_type in INSTANCE_TYPES:
+    try:
+        instance_type_module = importlib.import_module(
+            f"paasta_tools.{instance_type}_tools"
+        )
+    except ModuleNotFoundError:
+        # paasta_native defies the convention...
+        log.warning(
+            f"Cannot lookup k8s namespace no module called paasta_tools.{instance_type}_tools"
+        )
+        continue
+    INSTANCE_TYPE_TO_K8S_NAMESPACE[instance_type] = getattr(
+        instance_type_module, "KUBERNETES_NAMESPACE", "paasta"
+    )
+
+print(INSTANCE_TYPE_TO_K8S_NAMESPACE)
+
```

Comment:
It's interesting that we already have metainformation about namespaces in both `KUBERNETES_NAMESPACE` global variables for some things, and in `cr_id` functions for other things, see e.g. https://github.com/Yelp/paasta/blob/f39268ee424c6b7561eb532f3ba06b875b9cff55/paasta_tools/flink_tools.py#L152.  Can we maybe use `cr_id` everywhere instead of introducing new `KUBERNETES_NAMESPACE` variables?

It's also better to move `INSTANCE_TYPE_TO_K8S_NAMESPACE` outside of `paasta_secrets_sync.py`

Reply:
Interesting, I hadn't spotted that `cr_id` function. Good shout on location of `INSTANCE_TYPE_TO_K8S_NAMESPACE` too, I'll find somewhere more sensible